### PR TITLE
cow: always process KvMods to add OldData to mods when building StateDeltas

### PR DESCRIPTION
**Pull request by cce**
## Summary

Through debugging we noticed OldData was not set when it should be. Later we realized that roundCowState.deltas() was not always adding OldData to KvMods, and the early exit on empty sdeltas was the culprit.

## Test Plan

Existing tests should pass, maybe we should have new tests for this kind of thing.

### CODE REVIEW

**Review by jannotti — COMMENTED** [review]
Can we remove this entire guard clause? I don't see any point in exiting early. If those values are len == 0, the for loops should (not) run just as quickly.

**Comment by algorandskiy** [comment]
+1 to remove, empty blocks are rare :)
Need a unit test for this, and a MT-related test for this.

File: ledger/acctupdates_test.go [hunk 1]
```diff
@@ -1499,7 +1499,9 @@ func TestKVCache(t *testing.T) {
 		if i < kvCnt/kvsPerBlock {
 			for j := 0; j < kvsPerBlock; j++ {
 				name := fmt.Sprintf("%d", curKV)
-				kvMods[name] = ledgercore.KvValueDelta{Data: nil}
+				// needs an old data, else optimized away.
+				// if oldData = "" there is the best chance of a bug, so we use that
+				kvMods[name] = ledgercore.KvValueDelta{Data: nil, OldData: []byte("")}
```
Comment (algorandskiy) [inline ledger/acctupdates_test.go:1504]:
```suggestion
				kvMods[name] = ledgercore.KvValueDelta{Data: nil, OldData: []byte{}}
```

**Review by algorandskiy — APPROVED** [review]

**Review by michaeldiamant — APPROVED** [review]
As intended, I confirmed the added tests fail in `master` while passing in the PR.

@cce Thanks again for your efforts to isolate + find the issue!

**Pull request merged**